src/common/logging.h

```diff
@@ -46,13 +47,19 @@ namespace marian {
  *
  * @param ... Message text and variables
  */
-#define ABORT(...)                                                      \
-  do {                                                                  \
-    checkedLog("general", "critical", __VA_ARGS__);                     \
-    ::marian::logCallStack(/*skipLevels=*/0);                           \
-    std::cerr << "Aborted from " << FUNCTION_NAME << " in " << __FILE__ \
-              << ": " << __LINE__ << std::endl;                         \
-    std::abort();                                                       \
+#define ABORT(...)                                                             \
+  do {                                                                         \
+    auto logger = spdlog::get("general");                                      \
+    if(logger == nullptr)                                                      \
+      logger = createStderrLogger("general", "[%Y-%m-%d %T] Error: %v");       \
+    else                                                                       \
+      logger->set_pattern("[%Y-%m-%d %T] Error: %v");                          \
+    checkedLog("general", "critical", __VA_ARGS__);                            \
+    checkedLog("general", "critical", "Aborted from {} in {}:{}",              \
+               FUNCTION_NAME, __FILE__, __LINE__);                             \
+    logger->set_pattern("%v");                                                 \
+    checkedLog("general", "critical", marian::getCallStack(/*skipLevels=*/0)); \
+    std::abort();                                                              \
```

Comment:
This refactorization is not related to removing Boost dependency, right? Just a question.

Reply:
No, this was a general aesthetic thing.